src/compiler/declarationEmitter.ts

```diff
@@ -1593,7 +1593,10 @@ namespace ts {
                 emitTypeOfVariableDeclarationFromTypeLiteral(node);
             }
             else if (!hasModifier(node.parent, ModifierFlags.Private)) {
-                writeTypeOfDeclaration(node, node.type, getParameterDeclarationTypeVisibilityError);
+                // use the checker's type, not the declared type, for an initialized parameter (that isn't a parameter property)
+                const isInitializedParameter = node.initializer && !(getModifierFlags(node) & ModifierFlags.ParameterPropertyModifier);
+                const typeNode = isInitializedParameter ? undefined : node.type;

```

Comment:
so we treat `foo(x?: string = "")` different from `foo(x?:string)` ? 

i think with this the first will be `foo(x?: string | undefined)` and for the second `foo(x?: string)`. i would say both should be the same.